persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/EditTitleServiceTest.java

```diff
@@ -0,0 +1,182 @@
+
+        document = new Document();
+    }
+
+    @Test
+    public void editTitleTest() throws Exception {
+        String title = "new title";
+        document.addContent(new Element("mods", MODS_V3_NS)
+                .addContent(new Element("titleInfo", MODS_V3_NS)
+                        .addContent(new Element("title", MODS_V3_NS).setText("original title"))));
+        when(binaryObj.getBinaryStream()).thenReturn(convertDocumentToStream(document));
+
+        service.editTitle(agent, pid, title);
+
+        verify(operationsMessageSender).sendUpdateDescriptionOperation(anyString(), pidsCaptor.capture());
+        Collection<PID> pids = pidsCaptor.getValue();
+        assertEquals(pids.size(), 1);
+        assertTrue(pids.contains(pid));
+    }
+
+    @Test(expected = AccessRestrictionException.class)
+    public void insufficientAccessTest() throws Exception {
+        String title = "new title";
+        document.addContent(new Element("mods", MODS_V3_NS)
+                .addContent(new Element("titleInfo", MODS_V3_NS)
+                        .addContent(new Element("title", MODS_V3_NS).setText("original title"))));
+        when(binaryObj.getBinaryStream()).thenReturn(convertDocumentToStream(document));
+
+        doThrow(new AccessRestrictionException()).when(aclService)
+                .assertHasAccess(anyString(), eq(pid), any(AccessGroupSet.class), any(Permission.class));
+
+        service.editTitle(agent, pid, title);
+    }
+
+    @Test
+    public void noModsTest() throws Exception {
+        String title = "new title";
+        when(binaryObj.getBinaryStream()).thenReturn(null);
+
+        service.editTitle(agent, pid, title);
+
+        verify(operationsMessageSender).sendUpdateDescriptionOperation(anyString(), pidsCaptor.capture());
+        Collection<PID> pids = pidsCaptor.getValue();
+        assertEquals(pids.size(), 1);
+        assertTrue(pids.contains(pid));
+    }
+
+    @Test
+    public void noTitleInMods() throws Exception {
+        String title = "new title";
+        document.addContent(new Element("mods", MODS_V3_NS)
+                .addContent(new Element("language", MODS_V3_NS)
+                        .addContent(new Element("languageTerm", MODS_V3_NS)
+                                .setText("eng")
+                                .setAttribute("authority", "iso639-2b")
+                                .setAttribute("type", "code"))));
+        when(binaryObj.getBinaryStream()).thenReturn(convertDocumentToStream(document));
+
+        service.editTitle(agent, pid, title);
+
+        verify(operationsMessageSender).sendUpdateDescriptionOperation(anyString(), pidsCaptor.capture());
+        Collection<PID> pids = pidsCaptor.getValue();
+        assertEquals(pids.size(), 1);
+        assertTrue(pids.contains(pid));
+    }
+
+    @Test
+    public void multipleTitlesInMods() throws Exception {
```

Comment:
this one will be slightly harder to verify, but should be doable. `assertTitleValue` could either be written to search all titleInfo elements for the value (see `document.getRootElement().getChildren().stream().anyMatch(...)`), or you could handle this one as a separate case. But it is important for us to check that there are still two titles after the update.